src/components/useFetch.js

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react";
+
+export const useFetch = (url) => {
+    const [state, setState] = useState({items:null, isLoaded:true, hasErrors: false});
```

Comment:
I suggest you separate each state property into its own state variables because when you are calling setState you are overwriting the whole state and it's more error-prone.